test/mitmproxy/proxy/protocol/test_http2.py

```diff
@@ -90,10 +90,12 @@ class _Http2TestBase:
     @classmethod
     def setup_class(cls):
         cls.options = cls.get_options()
-        tmaster = tservers.TestMaster(cls.options)
-        tmaster.addons.add(core.Core())
-        cls.proxy = tservers.ProxyThread(tmaster)
+        cls.proxy = tservers.ProxyThread(tservers.TestMaster, cls.options)
         cls.proxy.start()
+        while True:
+            if cls.proxy.tmaster:
+                break
+            time.sleep(0.01)
```

Comment:
I'm seeing this block of code multiple times in this PR - maybe we should extract it into a helper function: `cls.proxy.wait_for_master()` or similar?

Reply:
Fixed.